Initial docs for Azure integration (#51631)
gravitational/teleport: docs/pages/admin-guides/teleport-policy/integrations/azure-sync.mdx

---
title: Discover Azure Access Patterns with Teleport Policy
description: Describes how to import and visualize Azure subscription access patterns using Teleport Policy and Access Graph.
---

Teleport Policy streamlines and centralizes access management across your entire infrastructure. You can view access
relationships in seconds, viewing unified, up-to-date relationships and policies between all users, groups, and
computing resources.

Teleport Policy with Access Graph offers insights into access patterns within your Azure subscription. By scanning
users, groups, service principals, and role definitions, it provides a visual representation and aids in enhancing the permission model
within your Azure environment. This functionality enables you to address queries such as:

- What resources are accessible to Azure users and roles?
- Which groups belong to other groups and how does that impact resource access?
- How broad or narrow is scoping across the subscription when assigning roles to users and groups?

> Utilizing the Access Graph to analyze permissions within an Azure subscription necessitates the setup of the Access
> Graph (AG) service, a Discovery Service, and integration with your Azure subscription.

Reviewer: ```suggestion
To use Access Graph to analyze permissions within an Azure subscription, you need to set up the Access
Graph (AG) service, Teleport Discovery Service, and an integration with your Azure subscription.
```
Style suggestion

Reviewer 2: Let's remove the (AG) as well. I don't think I've ever heard someone refer to it as "AG"

Access Graph is a feature of the [Teleport Policy](https://goteleport.com/platform/policy/) product that is available to Teleport Enterprise customers.

After logging in to the Teleport UI, go to the Management tab. If enabled, Access Graph options can be found under

> the Permission Management section.

Reviewer 2: The Teleport 17 UI doesn't have a Management tab.

We might get a higher quality piece of documentation if we pulled up the UI and went through the feature writing the guide from scratch rather than copying from an existing guide. When we just copy paste from other guides we also copy old and outdated info and create more work for ourselves to fix.

## How it works

Access Graph discovers Azure access patterns, synchronizes various Azure resources,
including users, groups, service principals, and role definitions.
These resources are then visualized using the graph representation detailed in the
[Teleport Policy usage page](../policy-how-to-use.mdx).

The importing process involves two primary steps:

### Polling Cloud APIs

The Teleport Discovery Service continuously scans the configured Azure subscriptions. At configurable intervals as
low as 15 minutes, it retrieves the following resources from your Azure account:

- Users
- Groups
- Service Principals
- Role Definitions
- Role Assignments
- Virtual Machines

> Once all the necessary resources are fetched, the Teleport Discovery Service pushes them to the Access Graph, ensuring

Reviewer: ```suggestion
Once all the necessary resources are fetched, the Teleport Discovery Service pushes them to Access Graph, ensuring
```

that it remains updated with the latest information from your Azure environment.

### Importing resources

> Teleport Policy’s Access Graph feature delves into the resources retrieved from your Azure subscription, crafting a
> graphical representation thereof.

Reviewer: ```suggestion
Teleport Policy’s Access Graph feature visualizes the resources retrieved from your Azure subscription.
```
Simplifying this sentence. We could think about removing the H3-level headings from "How it works", since this H3-level section is so short.

## Prerequisites

- A running Teleport Enterprise cluster v17.2.1 or later.
- Teleport Policy enabled for your account.
- For self-hosted clusters:
- Ensure that an up-to-date `license.pem` is used in the Auth Service configuration.
- A running Access Graph node v1.27.0 or later.
Check the [Teleport Policy page](../teleport-policy.mdx) for details on
how to set up Access Graph.
- The node running the Access Graph service must be reachable from the Teleport Auth Service.

## Step 1/2. Configure Discovery Service (Self-hosted only)

<Notice type="warning">
  If you have a managed Teleport Enterprise cluster, you can disregard

>   this step, as managed Teleport Enterprise already operates a properly configured

Reviewer 2: I really dislike this terminology. If we mean cloud let's just say cloud.

Every Teleport Enterprise cluster is _managed_ by someone. This is ambiguous.

Author: Good call--will update all docs at least in `docs/pages/admin-guides/teleport-policy/integrations/*` to use "cloud" or "cloud-hosted", just so they're consistent.

Author: FYI @ptgott since there's a mix of _managed_ and _cloud-hosted_ throughout the docs.

  Discovery Service within your cluster.
</Notice>

To activate the Teleport Discovery Service, add the provided snippet to your Auth Service configuration.
This service monitors dynamic `discovery_config` resources that are set up with the `discovery_group` matching
<Var name="access-graph-disc" />.

```yaml
discovery_service:
  enabled: true
  discovery_group: <Var name="access-graph-disc" />

> ```

Reviewer 2: This is a little confusing, because we say to add this snippet to your auth service configuration, but the Teleport config has both a top level `auth_service` and `discovery_service` section.

Without more context, how can we expect the user to know where to insert this snippet?

Notice that if you already operate a Discovery Service within your cluster,
it's possible to reuse it as long as the following requirements are met:

- On step 2, you match the `discovery_group` with the existing Discovery Service's
`discovery_group`.
- Access Graph service is reachable from the machine where Discovery Service runs.

## Step 2/2. Set up Access Graph Azure Sync

To configure the Teleport Discovery Service, the Azure managed identity running the Discovery service within
Azure must be given the right permissions to fetch Azure resources. Within the
[Azure Cloud Shell](https://portal.azure.com/#cloudshell/), download the
[Teleport binary](https://goteleport.com/download/#install-links) in `.tar.gz` format and use it to run the integration
command.

You'll need to specify the following in the commands/configuration below:
- <Var name="teleport-version"/>
- <Var name="azure-subscription-id"/>
- <Var name="azure-principal-id"/>

> - <Var name="azure-discovery-role"/>

Reviewer 3: Are these already set somewhere?

Author: I don't believe so. As in: have these variables been configured elsewhere in Teleport already?

Download and extract the Teleport binary:
```shell
curl -L https://cdn.teleport.dev/teleport-v<Var name="teleport-version"/>-linux-amd64-bin.tar.gz | tar xz

> ```

Author: Hmm, it's not really installation per se, but just downloading/extracting the Teleport binary to run a command. We wouldn't want to actually install Teleport for this particular step.

Run the integration command:
```shell
./teleport integration configure access-graph azure --principal-id <Var name="azure-principal-id"/> --role-name <Var name="azure-discovery-role"/> --subscription-id <Var name="azure-subscription-id"/>
```

Once the managed identity is configured with the correct permissions, add the following to the Discovery Service
configuration to enable it for fetching Azure resources:

```yaml
discovery_service:
  access_graph:
    azure:
      - subscription_id: <Var name="azure-subscription-id"/>
```

> The Discovery Service will now periodically fetch resources from your Azure subscription.

Reviewer 3: Is this required on Teleport Cloud?

Author: No. I'll actually create a separate section for this, since it's for self-hosted only.
